### ISSUE TITLE
Delivery Warehouse in Multi-Company is ignored in Sales Order

### ISSUE THREAD

**Issue by dedma** [frappe/erpnext#7880]
I seem to have encountered a bug when trying to create a Sales Order in a multi-company setup.

Here's a background:

I have a running ERPNext installation that is set to handle multiple companies. The main company acts as a distributor and sells it's times at wholesale to the other companies. The other companies then sell those purchased items to retail customers. So the item masters were set to have the default warehouse with the main distribution company (AR1).

So we go through the purchasing cycle of the distributor purchasing its stock from suppliers and getting them in the warehouse. We then proceed to the sales cycle where the distributor sells it's stock to the first reseller subcompany (BR1) and the items are moved out from AR1's warehouse for delivery. From BR1's side, their purchase cycle is completed and the items are now stocked in BR1's warehouse.

Now on to the issue:

Now BR1 initiates a sales cycle to sell the goods to retail customers. BR1 creates a Sales Order, adds the items in the sales order, indicates the "Delivery Warehouse" as Stores - BR1 which is where the items are stocked. But when saving the SO, an error is displayed: "Delivery Warehouse required for stock item".

Double checked the Stores - BR1 warehouse is indicated in "Delivery Warehouse" and items are indeed in stock in that warehouse. Tried again, same error: "Delivery Warehouse required for stock item". Couldn't save the SO with this persistent error.

Went to item master. Changed the "Default Warehouse" of the item from Stores - AR1 to Stores - BR1. Saved item changes. 

Proceeded to create an SO again as BR1, added the same items to the SO, double checked and made sure Delivery Warehouse was set to Stores - BR1. This time it saved. And got submitted successfully.

Now tried to create a new SO as AR1 to sell the same item again to BR1, this time AR1 is getting the error of "Delivery Warehouse required for stock item".

Went back to item master, removed the Default Warehouse item property and left it blank. Saved item settings.

Tried to create SO's for both AR1 and BR1, now both companies get the error "Delivery Warehouse required for stock item" when trying to save the newly created SO's.

In summary, it seems like the "Delivery Warehouse" field in the SO is being ignored and wants to persistently use the "Default Warehouse" that is indicated in the Item properties.

Any feedback or workaround?

**Comment by dedma** [comment]
Hoping you guys can provide feedback or a workaround.

Thanks

**Comment by dedma** [comment]
Any feedback on this item?

**Comment by dedma** [comment]
Has anyone else been able to replicate my issue?

**Comment by dedma** [comment]
Thank you @nabinhait for tagging this issue. Is there a projected timeline when version 8 is expected to be released?

**Comment by vishdha** [comment]
Step to Reproduce:
1. Created Purchase Order from AR1 to Test Supplier.
2. Created Purchase order from BR1 to AR1
3. Created Sales Order from AR1 to BR1
4. Created Purchase Order from For Test Company to BR1
5. Created Sales Order from BR1 to For Test Company

![anim](https://user-images.githubusercontent.com/6947417/29812111-317fa890-8cc3-11e7-829b-945f61ef7c1b.gif)
